src/main/java/net/fabricmc/loom/mixin/MixinMappingProviderTiny.java

```diff
@@ -110,34 +111,23 @@ public MappingField getFieldMapping(MappingField field) {
 
 	@Override
 	public void read(File input) throws IOException {
-		Mappings mappings;
-
-		try (FileInputStream stream = new FileInputStream(input)) {
-			mappings = MappingsProvider.readTinyMappings(stream, false);
-		}
-
-		for (ClassEntry entry : mappings.getClassEntries()) {
-			classMap.put(entry.get(from), entry.get(to));
+		TinyTree tree;
+		try (BufferedReader reader = new BufferedReader(new FileReader(input))) {
+			tree = TinyMappingFactory.loadWithDetection(reader);
 		}
 
-		for (FieldEntry entry : mappings.getFieldEntries()) {
-			EntryTriple fromEntry = entry.get(from);
-			EntryTriple toEntry = entry.get(to);
+		for (ClassDef cls : tree.getClasses()) {
+			String fromClass = cls.getName(from);
+			String toClass = cls.getName(to);
+			classMap.put(fromClass, toClass);
 
-			fieldMap.put(
-					new MappingField(fromEntry.getOwner(), fromEntry.getName(), fromEntry.getDesc()),
-					new MappingField(toEntry.getOwner(), toEntry.getName(), toEntry.getDesc())
-			);
-		}
-
-		for (MethodEntry entry : mappings.getMethodEntries()) {
-			EntryTriple fromEntry = entry.get(from);
-			EntryTriple toEntry = entry.get(to);
+			for (FieldDef field : cls.getFields()) {
```

Comment:
consider combining these 2 into one method that looks like this:
```
private void <T extends Descriptored> readDescriptored(List<T> descriptoredList, TriFunction<ClassEntry,String,Descriptor,T> constructor){
   for (T descriptored : descriptoredList) {
       ... 
   }
}
```

Reply:
i'd just like have 6 lines of code than having some callbacks. this is not that big a problem really

Reply:
I think that this way makes it easier to change later, but leaving it is fine too.